File: backend/app/Enums/SwipeDirection.php
<?php

namespace App\Enums;

/**
 * swipes.direction. `super` is reserved for the proposed Super Like
 * (docs/03 §Swipes, TBD-11) — writing rows is Phase 1 item 6, not item 5.
 */
enum SwipeDirection: string
{
    case Left = 'left';
    case Right = 'right';
    case Super = 'super';
}

File: mobile/lib/matching/domain/swipe_direction.dart
/// Mirrors `App\Enums\SwipeDirection` on the backend. `super` is accepted by
/// the API (matches the schema enum) but Super Like's proposed extras
/// (limits, premium gating, a distinct match celebration, the swipe-up
/// gesture) aren't built — [TBD-11], pending open decision #11 — so the UI
/// only ever sends `left`/`right`.
enum SwipeDirection {
  left('left'),
  right('right'),
  superLike('super');

  const SwipeDirection(this.apiValue);

  final String apiValue;
}
